core/GlobalState.cc

```diff
@@ -2279,12 +2279,16 @@ unique_ptr<LocalSymbolTableHashes> GlobalState::hash() const {
         counter++;
         // No fields are ignored in hashing.
         uint32_t symhash = field.hash(*this);
-        if (field.flags.isStaticField) {
+        if (field.flags.isStaticField && !field.flags.isStaticFieldTypeAlias && !field.isClassAlias()) {
             auto &target = staticFieldHashesMap[WithoutUniqueNameHash(*this, field.name)];
             target = mix(target, symhash);
+            uint32_t staticFieldShapeHash = field.fieldShapeHash(*this);
```

Comment:
Rename `fieldShapeHash` to `staticFieldShapeHash`?

(Related, it is very weird that instance variables, static fields, and several other things get stuffed into a single thing.)

Reply:
yeah it's odd. obviously it all just used to be on symbols, and there were no individual classes for various kinds of symbols' data. I have no context on why whoever factored out the symbol kinds (jvilk I think?) decided to put fields and static fields into one `Field` class. Field and static field definitely used to be identified as completely different things:

https://github.com/sorbet/sorbet/blob/e50896e71d02ff90abb4311e06298f0314cbb48d/core/Symbols.h#L63-L69

Reply:
I'll rename to staticFieldShapeHash in a follow up change.